library/std/src/sys/windows/path.rs

```diff
@@ -10,6 +10,49 @@ mod tests;
 pub const MAIN_SEP_STR: &str = "\\";
 pub const MAIN_SEP: char = '\\';
 
+/// A null-terminated `[u16]` string.
+#[unstable(feature = "fs_native_path_internals", issue = "none")]
+#[repr(transparent)]
+#[derive(Debug)]
+pub struct WindowsPath(pub [u16]);
```

Comment:
Why is this called `WindowsPath` instead of just `NativePath` (which is used for all other targets)?

Reply:
Looking back at the history here it seems I was originally thinking of making an equivalent of `CStr` (called `WStr` at the time). As the idea here evolved and narrowed in scope this eventually became `WindowsPath`,

But yes, I should just call this `NativePath` too. If there's a need for a more general type then that can and should be a separate PR.